src/adr/0018-polyglot-strategy.md

```diff
@@ -0,0 +1,158 @@
+
+Pros:
+* It's not a lot of work and yet it opens James to alternatives and can attract people outside Java developers community
+
+Cons:
+* Documentation without implementation often gets outdated when things move forward
+* We don't really gain knowledge on the polyglot matters as a community and won't be able to help users much
+
+### Option 3: Document the usage of polyglot components for some popular languages
+
+Pros:
+* It opens James to alternatives and can attract people outside Java developers community
+
+Cons:
+* Documentation without implementation often gets outdated when things move forward
+* For such complex subject it's probably harder to do than actually implementing a component in another language
+* We don't really gain knowledge on the polyglot matters as a community and won't be able to help users much
+
+### Option 4: Actually implement some mailets in some popular languages
+
+Pros:
+* It's probably not a lot of work, a mailet is just a simple class, probably easy to do in most JVM language
+* It makes us learn how it works and maybe will help us go further than the basic polyglot experience by doing small
+enhancement to the codebase
+* We can document the process and illustrate with some actual code
+* It opens James to alternatives and can attract people outside Java developers community
+
+Cons:
+* It makes the codebase more complex, requiring knowledge in another language
+* It can have a negative impact on the build time and dependencies download
+
+### Option 5: Actually implement some components in some popular languages
+
+Pros:
+* Leverage a modern language for some complex component
+* It makes us learn how it works and maybe will help us go further than the basic polyglot experience by doing small
+enhancement to the codebase
+* We can document the process and illustrate with some actual code
+* It opens James to alternatives and can attract people outside Java developers community
+
+Cons:
+* It makes the codebase more complex, requiring knowledge in another language
+* It can have a negative impact on the build time and dependencies download
+
+### Option I: Clojure
+
+Pros:
+* Functional Language
+
+Cons:
+* Weak popularity
+* No prior experience in the current team
+* Not statically typed hence less likely to fit the size of the project
+
+### Option II: Groovy
+
+Pros:
+* More advanced than current Java on most topics
+
+Cons:
+* No prior experience in the current team
+* Not very FP
+* Replaced in JVM community by Kotlin last years
+
+### Option III. Kotlin
+
+Pros:
+* Great Intellij support
+* Most of the good parts of Scala
+* FP-ish with Arrow
+* Coroutine for handling high-performance IOs
+
+Cons:
+* Lack of some FP constructs like proper Pattern Matching, persistent collections 
+* Despite progress done by Arrow, Kotlin community aims mostly at writing "better Java"
+
+#### Option IV. Scala
+
+Pros:
+* Rich FP community and ecosystem
+* Existing knowledge in the current team
+
```

Comment:
you can add that it is strongly statically typed so it's a good fit for the project size ?

Reply:
is it more typed than our use of Java? or Kotlin?